api/src/test/java/org/openmrs/util/ConceptReferenceRangeUtilityTest.java

```diff
@@ -0,0 +1,354 @@
+		person.setId(1);
+		
+		assertTrue(
+			conceptReferenceRangeUtility.evaluateCriteria(
+				"$patient.getAge() > 1 && $patient.getAge() < 10", 
+				person)
+		);
+	}
+
+	@Test
+	public void testAgeInMonths_shouldReturnTrueIfAgeIsWithinRange() {
+		calendar = Calendar.getInstance();
+		calendar.add(Calendar.MONTH, -5);
+		Date birthDate = calendar.getTime();
+		person.setBirthdate(birthDate);
+		
+		assertTrue(
+			conceptReferenceRangeUtility.evaluateCriteria(
+				"$patient.getAgeInMonths() > 1 && $patient.getAgeInMonths() < 12",
+				person)
+		);
+	}
+
+	@Test
+	public void testAgeInRange_shouldThrowExceptionIfCriteriaIsInvalid() {
+		calendar = Calendar.getInstance();
+		calendar.add(Calendar.YEAR, -1);
+		person.setBirthdate(calendar.getTime());
+
+		assertThrows(RuntimeException.class, () -> 
+			conceptReferenceRangeUtility.evaluateCriteria("invalidCriteria", person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldThrowAnExceptionIfCriteriaIsEmpty() {
+		calendar = Calendar.getInstance();
+		calendar.set(2019, Calendar.JUNE, 2);
+		person.setBirthdate(calendar.getTime());
+
+		IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
+			conceptReferenceRangeUtility.evaluateCriteria("", person);
+		});
+
+		assertTrue(exception.getMessage().contains("criteria is empty"));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfPersonAgeIsNotSet() {
+		assertFalse(
+			conceptReferenceRangeUtility.evaluateCriteria(
+				"$patient.getAge() >= 15 && $patient.getAge() <= 50",
+				person)
+		);
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnTrueIfGenderMatches() {
+		person.setGender("M");
+
+		assertTrue(conceptReferenceRangeUtility.evaluateCriteria("$patient.getGender().equals('M')", person));
+	}
+
+	@Test
+	public void testCriteriaWithPerson_shouldReturnTrueIfGenderMatches() {
+		person.setGender("M");
+
+		assertTrue(conceptReferenceRangeUtility.evaluateCriteria("$patient.getGender().equals('M')", person));
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnFalseIfGenderDoesNotMatch() {
+		person.setGender("F");
+
+		assertFalse(conceptReferenceRangeUtility.evaluateCriteria("$patient.getGender().equals('M')", person));
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnFalseIfGenderIsNull() {
+		assertFalse(conceptReferenceRangeUtility.evaluateCriteria("$patient.getGender().equals('M')", person));
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnTrueIfAgeAndGenderMatch() {
+		calendar = Calendar.getInstance();
+		calendar.add(Calendar.YEAR, -5);
+		person.setBirthdate(calendar.getTime());
+		person.setGender("M");
+
+		assertTrue(
+			conceptReferenceRangeUtility.evaluateCriteria(
+				"$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')", 
+				person)
+		);
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnTrueIfAgeOrGenderMatch() {
+		calendar = Calendar.getInstance();
+		calendar.add(Calendar.YEAR, -5);
```

Comment:
Can we make this test in such a way that changing `||` to `&&` would make it fail?